cmd/helpdesk-faq/main.go

```diff
@@ -0,0 +1,169 @@
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func validateOptions(o options) error {
+	_, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	return o.kubernetesOptions.Validate(false)
+}
+
+func router(kubeClient kubernetes.Interface) *http.ServeMux {
+	handler := http.NewServeMux()
+
+	handler.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		page := map[string]bool{"ok": true}
+		if err := json.NewEncoder(w).Encode(page); err != nil {
+			logrus.WithError(err).WithField("page", page).Error("failed to encode page")
+		}
+	})
+
+	handler.HandleFunc("/api/v1/faq-items", func(w http.ResponseWriter, r *http.Request) {
+		logrus.WithField("path", "/api/v1/faq-items").Info("serving")
+
+		configMap, err := getConfigMap(kubeClient)
+		if err != nil {
+			logrus.WithError(err).Fatal("unable to get helpdesk-faq config map")
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		page := Page{}
+		for _, item := range configMap.Data {
+			faqItem := &FaqItem{}
+			if err := json.Unmarshal([]byte(item), faqItem); err != nil {
+				logrus.WithError(err).Fatal("unable to unmarshall faq item")
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			page.Data = append(page.Data, *faqItem)
+		}
+
+		if callbackName := r.URL.Query().Get("callback"); callbackName != "" {
+			bytes, err := json.Marshal(page)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			w.Header().Set("Content-Type", "application/javascript")
+			template.JSEscape(w, []byte(callbackName))
+			if n, err := fmt.Fprintf(w, "(%s);", string(bytes)); err != nil {
+				logrus.WithError(err).WithField("n", n).Error("failed to write content")
+			}
+		} else {
+			w.Header().Set("Content-Type", "application/json")
+			if err := json.NewEncoder(w).Encode(page); err != nil {
+				logrus.WithError(err).WithField("page", page).Error("failed to encode page")
+			}
+		}
+	})
+
+	return handler
+}
+
+func getConfigMap(kubeClient kubernetes.Interface) (*v1.ConfigMap, error) {
+	configMap, err := kubeClient.CoreV1().ConfigMaps(ci).Get(context.TODO(), faqConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if configMap.Data == nil {
+		configMap.Data = make(map[string]string)
+	}
+
+	return configMap, nil
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed go gather options")
+	}
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+	level, _ := logrus.ParseLevel(o.logLevel)
+	logrus.SetLevel(level)
+
+	kubeClient, err := o.kubernetesOptions.ClusterClientForContext(appCIContextName, false)
```

Comment:
nit:
Make a flag of the tool for `dry-run` and use it as the 2nd param of the function?

Reply:
Ack. I hope we will keep it readonly as the tool grows. 😄